### ISSUE TITLE
Logging metrics (learning rate) to progress bar

### ISSUE THREAD

**Issue by jonashaag** [PyTorchLightning/pytorch-lightning#4047]
With pytorch-lightning >= 0.10.0 and `LearningRateMonitor`, the learning rate is automatically logged (using `logger.log_metric`). Is there a built-in way to log the learning rate to the tqdm progress bar?

Generally speaking, is there a built-in way to log metrics to the progress bar without modifying each of the callbacks?

**Comment by rohitgr7** [comment]
Try overriding this https://pytorch-lightning.readthedocs.io/en/latest/lightning_module.html#get-progress-bar-dict

**Comment by jonashaag** [comment]
Saw that, but where do I get the metrics from?

**Comment by rohitgr7** [comment]
I think it was `self.trainer.callback_metrics`. But here you need `learning_rate` so you can get that from `trainer.optimizers`

**Comment by jonashaag** [comment]
Thanks. I was wondering if there is a generic way to do it, i.e. a way to show non-LR metrics and LR on the progress bar, without having to change the metrics/callbacks code.

So yeah I guess I could add some "brute force checking" code to `training_step` that looks for the metric in different places.

**Comment by awaelchli** [comment]
> I was wondering if there is a generic way to do it, i.e. a way to show non-LR metrics and LR on the progress bar, without having to change the metrics/callbacks code.

There is. As mentioned by @rohitgr7 you can use 

```python 
self.log("my_lr", current_lr, prog_bar=True, on_step=True)
```
from your `training_step`. 

There is also a monitor that logs the learning rate to your logger. See [LearningRateMonitor](https://pytorch-lightning.readthedocs.io/en/latest/generated/pytorch_lightning.callbacks.LearningRateMonitor.html#pytorch_lightning.callbacks.LearningRateMonitor)

**Comment by awaelchli** [comment]
I do not recommend to log learning rates etc. to the progress bar. 1. these numbers are lost after you close the terminal 2. it's really nothing compared to a graph in which you can easily see the rate of change, compare to other runs etc.

**Comment by jonashaag** [comment]
Would you be open for a PR that documents how to log LR and other metrics to progbar?

**Comment by awaelchli** [comment]
Logging to the progress bar is covered here
https://pytorch-lightning.readthedocs.io/en/latest/logging.html#logging-from-a-lightningmodule
This is accessible in the left side bar under Optional extensions - Logging

Feel free to open a PR if you see a way to improve this section.

**Comment by Dehde** [comment]
I am interested in the same thing as @jonashaag. Unfortunately, your link above (https://pytorch-lightning.readthedocs.io/en/latest/logging.html#logging-from-a-lightningmodule) is broken. Where can I find this now?

**Comment by Dehde** [comment]
Think I found it: https://pytorch-lightning.readthedocs.io/en/stable/extensions/logging.html

**Comment by Dehde** [comment]
But I still don't know how to access the current learning rate
